website/content/docs/connect/proxies/integrate.mdx

```diff
@@ -7,6 +7,14 @@ description: >-
 
 # Custom Proxy Configuration for Service Mesh
 
+~> **Note:** The Connect Native Golang SDK and `v1/agent/connect/authorize`, `v1/agent/connect/ca/leaf`, 
+ and `v1/agent/connect/ca/roots` APIs are deprecated and will be removed in a future release. Although Connect Native 
+ is still supported we discourage new users from adopting this feature at this time, as we plan to remove the feature
+ when a future long term replacement such as proxyless GRPC via XDS is delivered (see [GH-10339](https://github.com/hashicorp/consul/issues/10339)).
+
+ The Native App Integration does not support many of the Consul's service mesh features, and is not under active development.
```

Comment:
```suggestion
 and `v1/agent/connect/ca/roots` APIs are deprecated and will be removed in a future release. Although connect native 
 is still supported we do not recommend leveraging this feature because it is deprecated and will be removed when the long term replacement, proxy-less gRPC connections to Consul xDS, is delivered. Refer to [GH-10339](https://github.com/hashicorp/consul/issues/10339) for additional information and to track progress toward the replacement functionality.

 The native app integration does not support many of the Consul's service mesh features, and is not under active development.
```